src/test/groovy/com/lesfurets/jenkins/TestHelperSingleton.groovy

```diff
@@ -0,0 +1,65 @@
+package com.lesfurets.jenkins
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+import com.lesfurets.jenkins.unit.HelperSingleton
+import com.lesfurets.jenkins.unit.PipelineTestHelper
+import org.junit.AfterClass
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Test
+
+import static com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.library
+import static com.lesfurets.jenkins.unit.global.lib.LocalSource.localSource
+import static org.assertj.core.api.Assertions.assertThat
+
+class TestHelperSingleton extends BasePipelineTest {
+
+    @BeforeClass
+    static void beforeClass() {
+        HelperSingleton.singletonInstance = new PipelineTestHelper()
+
+        String sharedLibs = this.class.getResource('/libs').getFile()
+
+        def library = library().name('commons')
+                .defaultVersion("master")
+                .allowOverride(true)
```

Comment:
I'm not sure how singleton would support backward compatibility. This may introduce a potential bug if singleton instance altered from previous test class would be still available to next test class. 

It means if we have TestA, TestB chain 
TestA register some method, TestB inherits this method via singleton instance   
If we change an order so TestB goes first it will throw an exception I suppose.

Reply:
Moreover this applicable for exactly test case.  
```
// what if I change the order commenting out some tests
// @Test
// void test_first() {}
@Test
void test_second() {}
```